### ISSUE TITLE
Add support for Subresource Integrity

### ISSUE THREAD

**Issue by frenzzy** [ztoben/assets-webpack-plugin#63]
Maybe we can add `integrity` information to final assets when [webpack-subresource-integrity](https://github.com/waysact/webpack-subresource-integrity) plugin is configured? For example:
```json
{
    "one": {
        "js": "/js/one_2bb80372ebe8047a68d4.bundle.js",
        "jsIntegrity": "sha256-gbZMWbP1x0a08t/M2/63KpSdL0vGPgExy4Hag678QxQ="
    },
    "two": {
        "js": "/js/two_2bb80372ebe8047a68d4.bundle.js",
        "jsIntegrity": "sha256-gbZMWbP1x0a08t/M2/63KpSdL0vGPgExy4Hag678QxQ="
    }
}
```

**Comment by DethAriel** [comment]
I see it's an old issue, could we get a heads-up from maintainers?

**Comment by ztoben** [comment]
This should be fixed in the latest version, let me know if not.

**Comment by edwardlai3582** [comment]
HI @ztoben , any example of using assets-webpack-plugin & webpack-subresource-integrity together? Appreciate it.
this is my current setting
```
    new SriPlugin({
      hashFuncNames: ['sha256'],
      enabled: true,
    }),
    new AssetsPlugin({
      path: path.resolve(__dirname, '../build'),
      filename: 'assets.json',
      prettyPrint: true,
    }),
```
It seems that the integrity info generated by SriPlugin is put in stats.compilation.assets and its not visible in compilation.getStats().toJson({....})

**Comment by ztoben** [comment]
@illl48 you're correct. I misunderstood how the SRI plugin works. I'll add an 'integrity' option to preserve their output.

**Comment by ztoben** [comment]
This should be available now in `v3.9.9`, using the `integrity` option.

```js
new AssetsPlugin({integrity: true})
```

Output should look like this:

```json
{
  "main": {
    "js": "/bundle.js",
    "jsIntegrity": "sha256-ANGwtktWN96nvBI/cjekdTvd0Dwf7SciIFTQ2lpTxGc= sha384-Ly439pF3K+J8hnhk1BEcjKnv1R9BApFYVIVJvr64PcgBjdT4N7hfPzQynItHwcaO"
  },
  "vendors~main": {
    "js": "/1.bundle.js",
    "jsIntegrity": "sha256-yqNi1hgeAdkXVOORgmVMeX+cbuXikoj6I8qWZjPegsA= sha384-4X75tnsGDwnwL5kBUPsx2ko9DeWy0xM8BcDQdoR185yho+OnxjjPXl2wCdebLWTG"
  }
}
```
